pkg/registry/common/recvfd/server.go

```diff
@@ -73,6 +78,10 @@ func (r *recvfdNseServer) Register(ctx context.Context, endpoint *registry.Netwo
 	}
 	returnedEndpoint = returnedEndpoint.Clone()
 
+	if endpointName != returnedEndpoint.Name {
```

Comment:
Curious, when would endpointName != returnedEndpoint.Name?

Reply:
I feel because remote side could set unique identity for incoming endpoint

Reply:
Got it.  I was curious if there was a specific use case in mind, or just an abundance of caution :)

Reply:
If they are same we are during heal Register, to update we alive. And if not we expect remote side to update with a proper name.